core/src/main/java/io/neow3j/protocol/core/response/Diagnostics.java

```diff
@@ -0,0 +1,199 @@
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.neow3j.types.Hash160;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Diagnostics {
+
+    /**
+     * The contracts that have been invoked in this invocation. The first hash is the script hash of the invocation
+     * script that is executed.
+     */
+    @JsonProperty("invokedcontracts")
+    private InvokedContract invokedContracts;
+
+    /**
+     * The resulting storage changes of the invocation.
+     */
+    @JsonProperty("storagechanges")
+    private List<StorageChange> storageChanges;
+
+    public Diagnostics() {
+    }
+
+    public Diagnostics(InvokedContract invokedContracts, List<StorageChange> storageChange) {
+        this.invokedContracts = invokedContracts;
+        this.storageChanges = storageChange;
+    }
+
+    public InvokedContract getInvokedContracts() {
+        return invokedContracts;
+    }
+
+    public List<StorageChange> getStorageChanges() {
+        return storageChanges;
+    }
+
+    public static class InvokedContract {
+
+        /**
+         * The script hash of the invoked contract.
+         */
+        @JsonProperty("hash")
+        private Hash160 hash;
+
+        /**
+         * The contracts that were invoked by this contract.
+         */
+        @JsonProperty("call")
+        private List<InvokedContract> invokedContracts = new ArrayList<>();
+
+        public InvokedContract() {
+        }
+
+        public InvokedContract(Hash160 hash) {
+            this(hash, new ArrayList<>());
+        }
+
+        public InvokedContract(Hash160 hash, List<InvokedContract> invokedContracts) {
+            this.hash = hash;
+            this.invokedContracts = invokedContracts;
+        }
+
+        public Hash160 getHash() {
+            return hash;
+        }
+
+        public List<InvokedContract> getInvokedContracts() {
+            return invokedContracts;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (!(o instanceof InvokedContract)) {
+                return false;
+            }
+            InvokedContract that = (InvokedContract) o;
+            return Objects.equals(getHash(), that.getHash()) &&
+                    Objects.equals(getInvokedContracts(), that.getInvokedContracts());
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(getHash(), getInvokedContracts());
+        }
+
+        @Override
+        public String toString() {
+            return "InvokedContract{" +
+                    "hash=" + hash +
+                    ", invokedContracts=" + invokedContracts +
+                    '}';
+        }
+
+    }
+
+    public static class StorageChange {
+
+        /**
+         * Represents the track state of the storage change.
```

Comment:
should it be "the trackED state"?

Reply:
There's an enum `TrackState` in the neo core, that's why I wrote it like that.

See here: https://github.com/neo-project/neo/blob/e7cc1fd047af74fd6126c0422102d067260d3a6c/src/neo/Persistence/TrackState.cs